Reorg Deploy section (#3048)
fishtown-analytics/docs.getdbt.com: website/docs/docs/deploy/deployment-tools.md

---
title: "Deploy with other tools"
id: "deployment-tools"
sidebar: "Deploy with other tools"
---

> Discover effortless ways to schedule and run your dbt jobs with the help of powerful tools like dbt Cloud, Airflow, Prefect, Dagster, automation server, or Cron. Use these to save time and increase efficiency by automating your data workflows, all while enjoying a hassle-free experience. 

Author: added steps first per gudiance from Leona

> Discover effortless ways to schedule and run your dbt jobs with the help of powerful tools like dbt Cloud, Airflow, Prefect, Dagster, automation server, or Cron. Use these to save time and increase efficiency by automating your data workflows, all while enjoying a hassle-free experience. 

Author: added steps first per gudiance from Leona

> Discover effortless ways to schedule and run your dbt jobs with the help of powerful tools like dbt Cloud, Airflow, Prefect, Dagster, automation server, or Cron. Use these to save time and increase efficiency by automating your data workflows, all while enjoying a hassle-free experience. 

Author: added steps first per gudiance from Leona

```suggestion
<TabItem value="adfsteps" label="dbt Cloud and ADF steps">
```

> Discover effortless ways to schedule and run your dbt jobs with the help of powerful tools like dbt Cloud, Airflow, Prefect, Dagster, automation server, or Cron. Use these to save time and increase efficiency by automating your data workflows, all while enjoying a hassle-free experience. 

Author: added steps first per gudiance from Leona

## Airflow

If your organization is using [Airflow](https://airflow.apache.org/), there are a number of ways you can run your dbt jobs, including:

<Tabs>

<TabItem value="airflowcloud" label="dbt Cloud">

Installing the [dbt Cloud Provider](https://registry.astronomer.io/providers/dbt-cloud) to orchestrate dbt Cloud jobs. This package contains multiple Hooks, Operators, and Sensors to complete various actions within dbt Cloud.

<Lightbox src="/img/docs/running-a-dbt-project/airflow_dbt_connector.png" title="Airflow DAG using DbtCloudRunJobOperator"/>
<Lightbox src="/img/docs/running-a-dbt-project/dbt_cloud_airflow_trigger.png" title="dbt Cloud job triggered by Airflow"/>

</TabItem>

<TabItem value="airflowcore" label="dbt Core">

Invoking dbt Core jobs through the [BashOperator](https://registry.astronomer.io/providers/apache-airflow/modules/bashoperator). In this case, be sure to install dbt into a virtual environment to avoid issues with conflicting dependencies between Airflow and dbt.

</TabItem>
</Tabs>

For more details on both of these methods, including example implementations, check out [this guide](https://www.astronomer.io/guides/airflow-dbt).

## Azure Data Factory

> Integrate dbt Cloud and [Azure Data Factory](https://learn.microsoft.com/en-us/azure/data-factory/) (ADF) for a smooth data process, from data ingestion to data transformation. You can seamlessly trigger dbt Cloud jobs upon completion of ingestion jobs by using the [dbt API](/docs/dbt-cloud-apis/overview) in ADF. The following video provides you with a detailed overview of how to trigger a dbt Cloud job via the API in ADF. 

Reviewer: ```suggestion
Integrate dbt Cloud and [Azure Data Factory](https://learn.microsoft.com/en-us/azure/data-factory/) (ADF) for a smooth data process, from data ingestion to data transformation. You can seamlessly trigger dbt Cloud jobs upon completion of ingestion jobs by using the [dbt API](/docs/dbt-cloud-apis/overview) in ADF. Need help building this out? [Contact us](https://www.getdbt.com/contact/) today!
```

> Need help building this out? - [Contact us](https://www.getdbt.com/contact/) today!

Reviewer: ```suggestion
```

> This video provides you with a detailed overview of how to trigger a dbt Cloud job via the API in Azure Data Factory.

Reviewer: ```suggestion
The following video provides you with a detailed overview of how to trigger a dbt Cloud job via the API in Azure Data Factory.
```

> 

Reviewer: ```suggestion
```

<LoomVideo id="8dcc1d22a0bf43a1b89ecc6f6b6d0b18" /> 

> 

Reviewer: ```suggestion
```

> **dbt Cloud + ADF steps**

Reviewer: I think these steps don't need an extra title.
```suggestion
```

---
title: "Deploy with other tools"
id: "deployment-tools"
sidebar: "Deploy with other tools"
---

> Discover effortless ways to schedule and run your dbt jobs with the help of powerful tools like [dbt Cloud](/docs/deploy/dbt-cloud-job), Airflow, Prefect, Dagster, automation server, Cron, or Azure Data Factory (ADF). 

Reviewer: Not sure if this captures it:
```suggestion
Discover other ways besides [dbt Cloud](/docs/deploy/dbt-cloud-job) to schedule and run your dbt jobs with the help of powerful tools like Airflow, Prefect, Dagster, automation server, Cron, or Azure Data Factory (ADF). 
```

Author: I've tweaked this slightly because I'd like people to know they can use the tools WITH dbt Cloud (trigger cloud jobs), as well as use the dbt Cloud scheduler.

## Airflow

If your organization is using [Airflow](https://airflow.apache.org/), there are a number of ways you can run your dbt jobs, including:

<Tabs>

<TabItem value="airflowcloud" label="dbt Cloud">

Installing the [dbt Cloud Provider](https://registry.astronomer.io/providers/dbt-cloud) to orchestrate dbt Cloud jobs. This package contains multiple Hooks, Operators, and Sensors to complete various actions within dbt Cloud.

<Lightbox src="/img/docs/running-a-dbt-project/airflow_dbt_connector.png" title="Airflow DAG using DbtCloudRunJobOperator"/>
<Lightbox src="/img/docs/running-a-dbt-project/dbt_cloud_airflow_trigger.png" title="dbt Cloud job triggered by Airflow"/>

</TabItem>

<TabItem value="airflowcore" label="dbt Core">

Invoking dbt Core jobs through the [BashOperator](https://registry.astronomer.io/providers/apache-airflow/modules/bashoperator). In this case, be sure to install dbt into a virtual environment to avoid issues with conflicting dependencies between Airflow and dbt.

</TabItem>
</Tabs>

For more details on both of these methods, including example implementations, check out [this guide](https://www.astronomer.io/guides/airflow-dbt).

> ## Azure Data Factory

Author: removed tabs per Katherine's feedback. replaced it with toggle instead

Integrate dbt Cloud and [Azure Data Factory](https://learn.microsoft.com/en-us/azure/data-factory/) (ADF) for a smooth data process, from data ingestion to data transformation. You can seamlessly trigger dbt Cloud jobs upon completion of ingestion jobs by using the [dbt API](/docs/dbt-cloud-apis/overview) in ADF. 

The following video provides you with a detailed overview of how to trigger a dbt Cloud job via the API in ADF. 

> Need help building this out? - [Contact us](https://www.getdbt.com/contact/) today!

Reviewer: I think this stray sentence might be better at the end of the 1st paragraph in line 38

<LoomVideo id="8dcc1d22a0bf43a1b89ecc6f6b6d0b18" /> 

> <details>

Reviewer: I think this works better outside of the <details>. They're hidden in there and not sure if screen readers would automatically read them that way. You might even put them before the video so you can get a glance at the steps covered in the video and get more from it when you watch it.

Author: ok gotcha, I've added it after the video per the original feedback from Katherine that the video was more detailed than the steps

## Airflow

If your organization is using [Airflow](https://airflow.apache.org/), there are a number of ways you can run your dbt jobs, including:

<Tabs>

<TabItem value="airflowcloud" label="dbt Cloud">

Installing the [dbt Cloud Provider](https://registry.astronomer.io/providers/dbt-cloud) to orchestrate dbt Cloud jobs. This package contains multiple Hooks, Operators, and Sensors to complete various actions within dbt Cloud.

<Lightbox src="/img/docs/running-a-dbt-project/airflow_dbt_connector.png" title="Airflow DAG using DbtCloudRunJobOperator"/>
<Lightbox src="/img/docs/running-a-dbt-project/dbt_cloud_airflow_trigger.png" title="dbt Cloud job triggered by Airflow"/>

</TabItem>

<TabItem value="airflowcore" label="dbt Core">

Invoking dbt Core jobs through the [BashOperator](https://registry.astronomer.io/providers/apache-airflow/modules/bashoperator). In this case, be sure to install dbt into a virtual environment to avoid issues with conflicting dependencies between Airflow and dbt.

</TabItem>
</Tabs>

For more details on both of these methods, including example implementations, check out [this guide](https://www.astronomer.io/guides/airflow-dbt).

> ## Azure Data Factory

Author: @kbrock91 - have added adf info here as written steps and your video too! let me know what you think!

For a more detailed overview of how to do this, refer to the following guidance:

<Tabs>

<TabItem value="adfsteps" label="dbt Cloud and ADF steps">

To use the dbt API to trigger a job in dbt Cloud through ADF:

1. In dbt Cloud, go to the job settings of the daily production job and turn off the scheduled run in the **Trigger** section.
2. You'll want to create a pipeline in ADF to trigger a dbt Cloud job.
3. Securely fetch the dbt Cloud service token from a key vault in ADF, using a web call as the first step in the pipeline.
4. Set the parameters in the pipeline, including the dbt Cloud account ID and  job ID, as well as the name of the key vault and secret that contains the service token. 
    * You can find the dbt Cloud job and account id in the URL, for example, if your URL is `https://cloud.getdbt.com/deploy/88888/projects/678910/jobs/123456`, the account ID is 88888 and the job ID is 123456
5. Trigger the pipeline in ADF to start the dbt Cloud job and monitor the status of the dbt Cloud job in ADF.
6. In dbt Cloud, you can check the status of the job and how it was triggered in dbt Cloud.

</TabItem>

<TabItem value="adfvideo" label="dbt Cloud and ADF video">

> This video provides you with a detailed overview of how to trigger a dbt Cloud job via the API once your ADF ingestion jobs finish.

Reviewer 2: This video provides you with a detailed overview of how to trigger a dbt Cloud job via the API in Azure Data Factory

<LoomVideo id="8dcc1d22a0bf43a1b89ecc6f6b6d0b18" /> 

</TabItem>
</Tabs>

   

> ## Prefect

Reviewer 3: @mirnawong1 can we make the two prefect headers consistent? i know this is old stuff but may as well do it now :)

Reviewer 3: meant to leave this on the H3s below
